src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,80 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        Class[] declaredClasses = interfaceClazz.getDeclaredClasses();
+        for (Class<?> declaredClass : declaredClasses) {
+            if (!declaredClass.isAnnotationPresent(Component.class)) {
+                throw new RuntimeException("Class " + declaredClass.getName()
+                        + " is not annotated with @Component");
+            }
```

Comment:
This loop is checking annotations on inner classes of the interface, which is not necessary for dependency injection. You should remove this loop as it does not relate to the task of injecting dependencies.